pandas/io/excel/_openpyxl.py

```diff
@@ -451,3 +462,280 @@ def write_cells(self, cells, sheet_name=None, startrow=0, startcol=0,
                             xcell = wks.cell(column=col, row=row)
                             for k, v in style_kwargs.items():
                                 setattr(xcell, k, v)
+
+
+class _OpenpyxlReader(_BaseExcelReader):
+
+    def __init__(self, filepath_or_buffer):
+        """Reader using openpyxl engine.
+
+        Parameters
+        ----------
+        filepath_or_buffer : string, path object or Workbook
+            Object to be parsed.
+        """
+        err_msg = "Install openpyxl >= 2.4.0 for Excel with Openpyxl support"
+
+        try:
+            import openpyxl
+        except ImportError:
+            raise ImportError(err_msg)
+        else:
+            if openpyxl.__version__ < LooseVersion('2.4.0'):
+                raise ImportError(err_msg +
+                                  ". Current version " + openpyxl.__version__)
+        super().__init__(filepath_or_buffer)
+
+    @property
+    def _workbook_class(self):
+        from openpyxl import Workbook
+        return Workbook
+
+    def load_workbook(self, filepath_or_buffer):
+        from openpyxl import load_workbook
+        return load_workbook(filepath_or_buffer, data_only=True)
+
+    @property
+    def sheet_names(self):
+        return self.book.sheetnames
+
+    @staticmethod
+    def _handle_usecols(frame, usecols):
+        column_names = frame.columns.values
+        if usecols:
+            _validate_usecols_arg(usecols)
+            usecols = sorted(usecols)
+            if any(isinstance(i, str) for i in usecols):
+                _validate_usecols_names(usecols, column_names)
+                frame = frame[usecols]
+            else:
+                frame = frame.iloc[:, usecols]
+        return frame
+
+    def _handle_sheet_name(self, sheet_name):
```

Comment:
Isn't this already handled in the base class `parse` method?

Reply:
Yes, somewhat. But see comment above for why I can't reuse the base class parser